lib/rubocop/cop/rails/pluralization_grammar.rb

```diff
@@ -0,0 +1,93 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for correct grammar when using active support's
+      # core extensions to the numeric classes
+      #
+      # @example
+      #   # bad
+      #   3.day.ago
+      #
+      #   # good
+      #   45.minutes.from_now
+      class PluralizationGrammar < Cop
+        DURATION_METHODS = [:second, :seconds,
+                            :minute, :minutes,
+                            :hour, :hours,
+                            :day, :days,
+                            :week, :weeks,
+                            :fortnight, :fortnights,
+                            :month, :months,
+                            :year, :years]
+
+        MSG = 'Prefer %s.%s for grammar reasons.'
+
+        def on_send(node)
+          receiver, method_name, *_args = *node
+          return if receiver.nil?
+          return unless duration_method?(method_name)
+          return unless literal_number?(receiver)
+          number, = *receiver
+          if singular_receiver?(receiver) && plural_method?(method_name)
+            add_offense(node,
+                        :expression,
+                        format(MSG, number, singularize(method_name)))
+          elsif plural_receiver?(receiver) && singular_method?(method_name)
+            add_offense(node,
+                        :expression,
+                        format(MSG, number, pluralize(method_name)))
+          end
+        end
+
+        private
+
+        def autocorrect(node)
+          lambda do |corrector|
+            method_name = node.loc.selector.source
+            replacement = if plural_method?(method_name)
+                            singularize(method_name)
+                          else
+                            pluralize(method_name)
+                          end
+            corrector.replace(node.loc.selector, replacement)
+          end
+        end
+
+        def plural_method?(method_name)
+          method_name.to_s.end_with?('s')
+        end
+
+        def singular_method?(method_name)
+          !plural_method?(method_name)
+        end
+
+        def singular_receiver?(receiver)
```

Comment:
I think that you could refactor this to take in the `number` instead of the `receiver`. The same would need to be done for `plural_receiver?`.